frontend/sentiment-dashboard/src/app/login/login.component.html

```diff
@@ -8,7 +8,16 @@ <h1>Login</h1>
     <label for="password">Password</label>
     <input id="password" class="form-control col-6" type="password" autocomplete="current-password" formControlName="password" placeholder="Password">
   </div>
-  <button class="btn btn-primary" type="submit">Login</button>
+  <div class="alert alert-danger col-6" role="alert" *ngIf="(loginForm.get('username').hasError('required') || loginForm.get('password').hasError('required')) && LoginForm.submitted">
```

Comment:
This seems like a pretty complex way to check if these inputs are required... It looks like you're already using Angular's validator system by attaching a `Validators.required` object to these form controls - is there a reason we can't just use `username.invalid` and `password.invalid`?